app/controllers/save_pages_controller.rb

```diff
@@ -18,6 +18,15 @@ def create
     redirect_to ensure_is_path(params[:page_path]) + "?personalisation=page_not_saved"
   end
 
+  def destroy
```

Comment:
Don't forget the unauthorized case here

Reply:
A user might have a session header (so they pass the `authenticate` `before_action`) but that header is invalid (so there's a 401 response).

Reply:
In fact, you could just drop the `before_action`.